onnxruntime/test/contrib_ops/quant_gemm_test.cc

```diff
@@ -35,7 +35,7 @@ void RunQuantGemmU8X8Test(const int M,
 
   constexpr int overflow_adjust = std::is_signed_v<WeightType> ? 2 : 1;
   constexpr int random_B_min = std::numeric_limits<WeightType>::min() / overflow_adjust;
```

Comment:
I don't understand this part. If it is already a min value, why do you divide it to something? 
For example, for float, the min value is  1.17549e-38. If you divide it by 2, what you get?

Reply:
And why this test only failed on ARM64?

Reply:
The adjust only happens when WeightType is int8. It adjusts the range to [-64, 63] to avoid overflow on AVX2 and AVX512 without VNNI. 

The bug/typo makes B is always -64 and zero point of B is -64 too, which leads to the scale of re-quantization to 0.f. Quant_value = float_vlaue/scale + zeropoint. ARM64 and x64 has different logic to handle divide by zero. That's why only arm has the issue.